packages/libs/components/src/plots/BipartiteNetwork.tsx

```diff
@@ -112,65 +131,165 @@ function BipartiteNetwork(
+            y: svgStyles.topPadding + svgStyles.nodeSpacing * indexInPartition,
+            labelPosition:
+              partitionIndex === 0 ? 'left' : ('right' as LabelPosition),
+            ...node,
+          };
+        });
+        return partitionWithCoordinates;
+      }),
+    [
+      column1Position,
+      column2Position,
+      data.partitions,
+      nodesByPartition,
+      svgStyles.nodeSpacing,
+      svgStyles.topPadding,
+    ]
   );
 
   // Assign coordinates to links based on the newly created node coordinates
-  const linksWithCoordinates = data.links.map((link) => {
-    const sourceNode = nodesByPartitionWithCoordinates[0].find(
-      (node) => node.id === link.source.id
-    );
-    const targetNode = nodesByPartitionWithCoordinates[1].find(
-      (node) => node.id === link.target.id
-    );
-    return {
-      ...link,
-      source: {
-        x: sourceNode?.x,
-        y: sourceNode?.y,
-        ...link.source,
-      },
-      target: {
-        x: targetNode?.x,
-        y: targetNode?.y,
-        ...link.target,
-      },
-    };
-  });
+  const linksWithCoordinates = useMemo(
+    () =>
+      data.links.map((link) => {
+        const sourceNode = nodesByPartitionWithCoordinates[0].find(
+          (node) => node.id === link.source.id
+        );
+        const targetNode = nodesByPartitionWithCoordinates[1].find(
+          (node) => node.id === link.target.id
+        );
+        return {
+          ...link,
+          source: {
+            x: sourceNode?.x,
+            y: sourceNode?.y,
+            ...link.source,
+          },
+          target: {
+            x: targetNode?.x,
+            y: targetNode?.y,
+            ...link.target,
+          },
+        };
+      }),
+    [data.links, nodesByPartitionWithCoordinates]
+  );
 
   const plotRect = plotRef.current?.getBoundingClientRect();
   const imageHeight = plotRect?.height;
   const imageWidth = plotRect?.width;
 
+  const nodes = useMemo(
+    () =>
+      nodesByPartitionWithCoordinates[0]
+        .concat(nodesByPartitionWithCoordinates[1])
+        .map((node) => ({
+          ...node,
+          actions: getNodeActions?.(node.id),
+        })),
+    [getNodeActions, nodesByPartitionWithCoordinates]
+  );
+
+  const activeNode = nodes.find((node) => node.id === activeNodeId);
+
+  useEffect(() => {
+    const element = document.querySelector('.bpnet-plot-container');
+    if (element == null) return;
+
+    element.addEventListener('click', handler);
+
+    return () => {
+      element.removeEventListener('click', handler);
+    };
+
+    function handler() {
+      setActiveNodeId(undefined);
+    }
+  }, [containerClass]);
+
   return (
     <>
       <div
         className={containerClass}
         style={{ width: '100%', ...containerStyles, position: 'relative' }}
       >
-        <div ref={plotRef} style={{ width: '100%', height: '100%' }}>
+        <div style={{ display: 'inline-flex' }}>
+          <Banner
```

Comment:
You beat me to the Banner haha! Two thoughts:
1. I think the banner might look nicer at the viz level so that it can float above the legends. We can add a little info icon, too. Alternatively i think centering it in the plot could help as well.
2. The dropdown is only available for eigengene nodes in plasmo/host. The other nodes won't have a dropdown for now. Maybe this should be handled in my PR, but would be easier to control the banner verbiage from the viz i'd think

Reply:
Oh wait i see there is `getNodeActions &&` on line 245... that should do it i'd think, but i'm still seeing the additional text in mbio. I need to look at this more

Reply:
I think I agree that this should go in the viz

Reply:
done